pkg/build/node/node.go

```diff
@@ -496,19 +497,88 @@ func (c *BuildContext) prePullImages(dir, containerID string) error {
 		return err
 	}
 
-	// preload images into containerd
-	// TODO(bentheelder): we can skip the move and chown steps if we go directly to this
-	if err = inheritOutputAndRun(cmder.Command(
-		"bash", "-c",
-		`containerd & find /kind/images -name *.tar -print0 | xargs -0 -n 1 -P $(nproc) ctr --namespace=k8s.io images import --no-unpack && kill %1 && rm -rf /kind/images/*`,
-	)); err != nil {
-		log.Errorf("Image build Failed! Failed to load images into containerd %v", err)
+	// setup image importer
+	importer := newContainerdImporter(cmder)
+	if err := importer.Prepare(); err != nil {
+		log.Errorf("Image build Failed! Failed to prepare containerd to load images %v", err)
+		return err
+	}
+
+	// TODO: return this error?
+	defer func() {
+		if err := importer.End(); err != nil {
+			log.Errorf("Image build Failed! Failed to tear down containerd after loading images %v", err)
+		}
+	}()
+
+	// get image tag fixing function for this version
+	fixRepository := repositoryCorrectorForVersion(ver)
+
+	// create a plan of image loading
+	loadFns := []func() error{}
+	for _, image := range c.bits.ImagePaths() {
+		image := image // capture loop var
+		loadFns = append(loadFns, func() error {
+			f, err := os.Open(image)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			//return importer.LoadCommand().SetStdout(os.Stdout).SetStderr(os.Stderr).SetStdin(f).Run()
+			// we will rewrite / correct the tags as we load the image
+			if err := exec.RunWithStdinWriter(importer.LoadCommand().SetStdout(os.Stdout).SetStderr(os.Stdout), func(w io.Writer) error {
+				return docker.EditArchiveRepositories(f, w, fixRepository)
+				// _, err := io.Copy(w, f)
+				// return err
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+	}
+
+	// run all image loading concurrently until one fails or all succeed
+	if err := concurrent.UntilError(loadFns); err != nil {
+		log.Errorf("Image build Failed! Failed to load images %v", err)
 		return err
 	}
 
 	return nil
 }
 
+func repositoryCorrectorForVersion(kubeVersion *version.Version) func(string) string {
```

Comment:
this logic should be leveraged to determine which tags we built as well when deciding what to pull.

Reply:
done.